tests/ccgbugs/t15623.nim

```diff
@@ -0,0 +1,8 @@
+block:
+  if false:
+    discard cast[ptr int](nil)[]
```

Comment:
(i had edited my previous comment too late)

  discard cast[ptr int](nil)[]
=>
  let s = cast[ptr int](nil)[]

(`discard` already worked before PR)

Reply:
I will revert this test now.